### PR TITLE
Use Argparse for FSL input/outputs from commands

### PR DESCRIPTION
**Pull request by cyril-data**
Add the lib argparse to interpret command in fsl files

### CODE REVIEW

**Comment by cmaumet** [comment]
@hermann74 can you review this PR?

File: bids_prov/fsl/fsl_parser.py [hunk 1]
```diff
@@ -138,75 +138,210 @@ def readlines(filename: str) -> Mapping[str, List[str]]:
...
+    parser = argparse.ArgumentParser(
+        add_help=False, conflict_handler='resolve')
+
+    inputs_kwarg = []
+    outputs_kwarg = []
+    parameters_value = []
+    parameters_no_value = []
+
+    if "used" in parameters:
+        parser, inputs_kwarg = _get_kwarg(parser, parameters["used"])
```
Comment (hermann74) [inline bids_prov/fsl/fsl_parser.py:265]: parser.add_argument outside function `_get_kwarg`

File: bids_prov/fsl/fsl_parser.py [hunk 2]
```diff
@@ -247,12 +383,15 @@ def build_records(groups: Mapping[str, List[str]], agent_id: str):
             for df in description_functions:
                 if df["name"] == command_name_end:
                     function_in_description_functions = True
-                    if "used" in df:
-                        entities, _ = get_entities(cmd_s, df["used"])
-                        inputs.extend(entities)
-                    if "generatedBy" in df:
-                        entities, _ = get_entities(cmd_s, df["generatedBy"])
-                        outputs.extend(entities)
+                    inputs, outputs, parameters = get_entities(cmd_s[1:], df)
```
Comment (hermann74) [inline bids_prov/fsl/fsl_parser.py:386]: not explicit that df called in get_entities is the one from command_name_end. Perhaps add and explicit variable name

File: bids_prov/afni/afni_parser.py [hunk 3]
```diff
@@ -44,7 +44,7 @@
 )
 
 
-def clean_label_suffix(label:str) ->str:
+def clean_label_suffix(label: str) -> str:
     """ Erase suffix like tlrc in label to keep link of passed entities label
```
Comment (hermann74) [inline bids_prov/afni/afni_parser.py:48]: more cleaning instructions in other branch dev-afni (PR 99). To be merged after

**Review by hermann74 — APPROVED** [review]
Good Job :+1:  @cyril-data  ! These change on the way we parse command lines in fsl seem to be more efficient. All functions parsed must be well described in description_function.json with inputs, ouputs and parameters. 
Nidm example on fsl are more deeper !  
Some little changes could render code more readable.

File: bids_prov/fsl/fsl_parser.py [hunk 4]
```diff
@@ -138,75 +138,210 @@ def readlines(filename: str) -> Mapping[str, List[str]]:
 #         return conf.bosh_config[key]
 #     return None
 
+def _get_kwarg(parser, serie,  with_value=True):
```
[the remark this answers is not in the document]
Comment (cyril-data) [reply bids_prov/fsl/fsl_parser.py]: Fix in last commit **answer hermann** 

image

File: bids_prov/fsl/fsl_parser.py (hunk 1 above)
Comment (cyril-data) [reply bids_prov/fsl/fsl_parser.py]: Fix in commit **answer hermann**

File: bids_prov/afni/afni_parser.py (hunk 3 above)
Comment (cyril-data) [reply bids_prov/afni/afni_parser.py:48]: Done with the merge

File: bids_prov/fsl/fsl_parser.py (hunk 2 above)
Comment (cyril-data) [reply bids_prov/fsl/fsl_parser.py]: FIX in commit **answer hermann **

image

**Comment by cmaumet** [comment]
@cyril-data: if you have addressed all the comments by Hermann (which I think is the case), I beleive this is good to merge!

**Review by cmaumet — CHANGES_REQUESTED** [review]
Hi @cyril-data! This looks very good to me. I only have spotted a few files at the root of the folder that I think should be deleted. Can you check? Once this is solved this is good to merge.

File: afni_default_proc.jsonld [hunk 5]
```diff
@@ -0,0 +1,571 @@
+{
```
[the remark this answers is not in the document]
Comment (cyril-data) [reply afni_default_proc.jsonld:1]: ok, thanks for your attention and sorry for the unwanted files. I check

[the remark this answers is not in the document]
Comment (cyril-data) [reply afni_default_proc.jsonld]: Now it should be good with #[4fb57aa](https://github.com/bids-standard/BEP028_BIDSprov/pull/98/commits/4fb57aa5b19c25426e2243036b49796e7e13b60b)

**Pull request merged**